src/components/pages/RootPage.tsx

```diff
@@ -1,49 +1,222 @@
-import {FC} from "react";
+import {FC, useMemo, useState} from "react";
 import useColumnService from "@/hooks/useColumnService.ts";
 import ColumnElement from "@/components/ui/column/ColumnElement.tsx";
 import {columnService} from "@/services/column/column.service.ts";
 import {ColumnView, CreateColumnDto} from "@/services/column/column.types.ts";
 import usePopup from "@/hooks/usePopup.tsx";
 import CreateColumnForm from "@/components/ui/column/create/CreateColumnForm.tsx";
 
+import {
+    DndContext,
+    DragEndEvent,
+    DragOverEvent,
+    DragOverlay,
+    DragStartEvent,
+    PointerSensor,
+    useSensor,
+    useSensors
+} from "@dnd-kit/core";
+
+import {arrayMove, SortableContext} from "@dnd-kit/sortable";
+import {createPortal} from "react-dom";
+import {ItemTypes} from "@/utils/id/ItemTypes.ts";
+import {TaskLookupView} from "@/services/task/task.types.ts";
+import TaskElement from "@/components/ui/task/TaskElement.tsx";
+import useTasksService from "@/hooks/useTasksService.ts";
+import styles from "./RootPage.module.scss";
+import {getColumnId, parseId} from "@/utils/id/id.utils.ts";
+
 const RootPage: FC = () => {
-    const {addColumn, columns, removeColumn, updateColumn} = useColumnService()
-    const {reversePopup, closePopup, Popup} = usePopup()
+    const {addColumn, columns, removeColumn, updateColumn, setColumns} = useColumnService();
+    const {tasks, setTasks, updateTask, removeTask} = useTasksService()
+    const {reversePopup, closePopup, Popup} = usePopup();
+    const columnsId = useMemo(() => columns.map((col) => getColumnId(col.id)), [columns]);
+    const [activeColumn, setActiveColumn] = useState<ColumnView | null>(null);
+    const [activeTask, setActiveTask] = useState<TaskLookupView | null>(null);
+
+    const sensors = useSensors(useSensor(PointerSensor, {
+        activationConstraint: {
+            distance: 3
+        },
+    }))
 
     const onSubmit = (data: CreateColumnDto) => {
         if (data.columnName.trim() !== "") {
-            columnService.create(data).then(data =>
-                addColumn(data)
-            )
+            columnService.create(data).then((data) => addColumn(data));
```

Comment:
Remove column and task service from root component and move this operations to hook